### ISSUE TITLE
osx-wifi-cli scan turns off the wifi when connected to a wifi network

### ISSUE THREAD

**Issue by pulkitmaloo** [danyshaanan/osx-wifi-cli#3]
I expected that even when connected to a network, it should scan and show the other available wifi networks. However, it turns off the wifi.

OS - macOS High Sierra 10.13.6

**Comment by danyshaanan** [comment]
Thanks for submitting an issue!

I have never encountered that. (Currently Sierra 10.12.5). Can  you post your command and the output? Also, can you do the same with ```/System/Library/PrivateFrameworks/Apple80211.framework/Versions/A/Resources/airport scan```? (That's pretty much all ```osx-wifi-cli scan``` does).

**Comment by pulkitmaloo** [comment]
`osx-wifi-cli scan` turns off my wifi, however `/System/Library/PrivateFrameworks/Apple80211.framework/Versions/A/Resources/airport scan` shows me a list of wifi networks as expected.

**Comment by danyshaanan** [comment]
Can you copypaste or screenshot your terminal when running these commands?

**Comment by danyshaanan** [comment]
If anyone can reproduce and supply more details, please reopen.

**Comment by mixn** [comment]
Hi @danyshaanan. :) Thanks for taking the time to create this, it seems like a very useful cli. 🎉 

It does behave a bit oddly for me though, so I thought I’d chime in here.

`osx-wifi-cli` ➝ Turns Wi-Fi off
`osx-wifi-cli pass` ➝ Turns Wi-Fi off
`osx-wifi-cli scan` ➝ Turns Wi-Fi off
`osx-wifi-cli on` ➝ Turns Wi-Fi on ✅ 
`osx-wifi-cli off` ➝ Gives what you’d usually expect from `--help`

![screen shot 2018-09-17 at 14 59 42](https://user-images.githubusercontent.com/672237/45624315-5343b700-ba8a-11e8-8654-fe1850136b67.png)

`osx-wifi-cli restart` ➝ Turns Wi-Fi off
`osx-wifi-cli NETWORK PASSWORD` ➝ Turns Wi-Fi off

**None** of the commands above give me feedback, other than the mentioned `osx-wifi-cli off`.

Running `/System/Library/PrivateFrameworks/Apple80211.framework/Versions/A/Resources/airport scan` (when connected to Wi-Fi) does indeed give me a list of scanned Wi-Fi networks.

Here’s a screenshot of running all commands. I’m using [manage-wifi-cli](https://github.com/sindresorhus/manage-wifi-cli) in between every step to reconnect to the network (as most commands turn it off).

I’m on macOS `10.12.6`. 

Hope that helps. 🙂

**Comment by danyshaanan** [comment]
Hi @mixn, and thanks for the details!

I have a few possible fixes, and I'd really appreciate it if you could clone the repo and run the following commands:
```
git clone git@github.com:danyshaanan/osx-wifi-cli.git
cd osx-wifi-cli
npm i
./osx-wifi-cli.js
./osx-wifi-cli.js off
./osx-wifi-cli.js on
git checkout literalOff
./osx-wifi-cli.js
./osx-wifi-cli.js off
./osx-wifi-cli.js on
```

The master branch includes a full rewrite, and the literalOff branch includes an additional test and output. I believe that this is all a single problem with the 'off' command, which I could understand with the output of all of the above.

Thanks!

**Comment by danyshaanan** [comment]
Disregard that. Problem is with node 10. Fixing now.

**Comment by danyshaanan** [comment]
@mixn Disregard that. Problem is with node 10. Fixing now.

**Comment by danyshaanan** [comment]
@mixn @pulkitmaloo - Fixed in version 2.0.0. (commit b9930374e565c718d2e3a83f9bc5ea3bf9e2dfbc). `npm -g i osx-wifi-cli` to fix. Thanks!

**Comment by mixn** [comment]
@danyshaanan Just upgraded and it works flawlessly / as expected now! 🎉 Thanks a ton!

Also, check out [meow](https://github.com/sindresorhus/meow), you might like it. :)
